cmd/otel-allocator/allocation/consistent_hashing.go

```diff
@@ -0,0 +1,189 @@
+package allocation
+
+import (
+	"fmt"
+	"github.com/buraksezer/consistent"
+	"github.com/cespare/xxhash/v2"
+	"github.com/go-logr/logr"
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/diff"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/url"
+	"sync"
+)
+
+var _ Allocator = &consistentHashingAllocator{}
+
+const consistentHashingStrategyName = "consistent-hashing"
+
+type hasher struct{}
+
+func (h hasher) Sum64(data []byte) uint64 {
+	return xxhash.Sum64(data)
+}
+
+type consistentHashingAllocator struct {
```

Comment:
A lot of this code looks _very_ similar to the least weighted strategy. It may be worth investigating in the future if we can combine these as I tried to do in #1068